### PR TITLE
Fix MixHandler to restore original loss function of the learner

### PR DESCRIPTION
**Pull request by roboserg**
If MixUp (derived from MixHandler) is assigned as a callback during learner instantiation, running lr_find() will cause MixUp/MixHandler to change the loss function of the learner without restoring it back. This will lead to an error during training (as discussed in the #fastbook-and-courses Discord channel).

This PR solves this issue by restoring the original loss function of the learner in the after_cancel_train event, which will be invoked by lr_find().

### CODE REVIEW

File: fastai/callback/mixup.py [hunk 1]
```diff
@@ -25,6 +25,9 @@ def before_train(self):
     def after_train(self):
         if self.stack_y: self.learn.loss_func = self.old_lf
 
+    def after_cancel_train(self):
+        if self.stack_y: self.learn.loss_func = self.old_lf
```
Comment (jph00) [inline fastai/callback/mixup.py:29]:
```suggestion
        self.after_train()
```

**Review by jph00 — CHANGES_REQUESTED** [review]
One little suggestion

File: fastai/callback/mixup.py (hunk 1 above)
Comment (roboserg) [reply fastai/callback/mixup.py]: yep, much cleaner, committed below.

**Comment by jph00** [comment]
Many thanks!
